packages/wdio-jasmine-framework/src/reporter.js

```diff
@@ -87,10 +90,31 @@ export default class JasmineReporter {
             })
         }
 
+        // errors on beforeAll are swallowed by Jasmine and can only be accessed later on
+        let errorsBeforeAll
+        let errorsAfterAll
+        if (suite.failedExpectations.length) {
+            let errors = suite.failedExpectations
+
+            if (this.shouldCleanStack) {
+                errors = suite.failedExpectations.map(x => this.cleanStack(x))
+            }
+
+            errorsBeforeAll = errors.filter(error => error.stack.includes('UserContext.beforeAll'))
+            errorsAfterAll = errors.filter(error => error.stack.includes('UserContext.afterAll'))
+        }
+
+        this.emitHookEnd(suite, 'before all', errorsBeforeAll)
```

Comment:
I'd rename hooks to beforeAll / afterAll to avoid possible overlapping with mocha hooks

Reply:
This was actually intentional. By going with this approach, the other reporters need less adjustment. For example, Allure shows the hooks in the report only if it matches this name exactly (since it matches for it). Since mocha and jasmine cannot be used together anyways, I don't think there would be any issues.